### ISSUE TITLE
⚗️ Investigate if we need to implement the noise XK handshake for libp2p

### ISSUE THREAD

**Issue by D4nte** [comit-network/comit-rs#651]
rust-libp2p provides noise but not the XK handshake. Need to investigate whether we still need/want that one.

Child of #744 
Blocked by #800.

**Comment by thomaseizinger** [comment]
Given the decision in #779, the scope of this changes to implementing the desired XK handshake in libp2p.

**Comment by D4nte** [comment]
@thomaseizinger could you change the scope in the body then please ? :)

**Comment by D4nte** [comment]
Not needed now that we understand how libp2p works.
